src/harness/sas_testcase.py

```diff
@@ -45,3 +53,50 @@ def assertValidResponseFormatForApprovedGrant(self, grant_response):
     # operationParam should not be set if grant is approved
     self.assertFalse('operationParam' in grant_response)
     self.assertTrue(grant_response['channelType'] in ('PAL', 'GAA'))
+
+  def assertRegistered(self, registration_request,
+                       conditional_registration_data=None):
+    for device in registration_request:
+      self._sas_admin.InjectFccId({'fccId': device['fccId']})
+    if conditional_registration_data:
+      self._sas_admin.PreloadRegistrationData(conditional_registration_data)
+
+    request = {'registrationRequest': registration_request}
+    response = self._sas.Registration(request)['registrationResponse']
+
+    # Check the registration response; collect CBSD IDs
+    cbsd_ids = []
+    for resp in response:
+      self.assertEqual(resp['response']['responseCode'], 0)
+      self.assertTrue('cbsdId' in resp)
+      cbsd_ids.append(resp['cbsdId'])
+
+    # Return list of cbsd_ids
+    return cbsd_ids
+
+  def assertRegisteredAndGranted(self, registration_request, grant_request,
+                                 conditional_registration_data=None):
+    self.assertEqual(len(registration_request), len(grant_request))
+    cbsd_ids = self.assertRegistered(registration_request,
+                                     conditional_registration_data)
+    list_of_cbsd_ids_and_grant_ids = []
+
+    for cbsd_id, grant_req in zip(cbsd_ids, grant_request):
+      grant_req['cbsdId'] = cbsd_id
+
+    request = {'grantRequest': grant_request}
+    grant_response = self._sas.Grant(request)['grantResponse']
+
+    # Check the grant response; collect CBSD and Grant IDs
+    for cbsd_id, grant_resp in zip(cbsd_ids, grant_response):
```

Comment:
Similar to my comment above, we might not want to use this helper function in the GRA_* testcases since those will require more careful checking of the grant response. We could call assertValidResponseFormatForApprovedGrant() from within assertRegisteredAndGranted(), which may mean we can use it in a few more testcases (e.g. GRA_20); this approach has the pros and cons listed in my earlier message. I slightly prefer not calling it inside the helper function.

In general, I think we should assume that the helper functions will take care of the common case and that any uncommon case will not necessarily be able to use them.